server/apiRoutes/comments.js

```diff
@@ -0,0 +1,127 @@
+const router = require('express').Router();
+const uuid = require('uuid/v4');
+const { Comment } = require('../db/index');
+const { isLoggedIn } = require('../../utils/backend');
+
+// type: post / event
+router.get('/:type/:id', async (req, res, next) => {
+  try {
+    let { offset } = req.query;
+    let { type, id } = req.params;
+
+    if (!type || !id) {
+      res.sendStatus(400);
+    }
+
+    offset = offset ? parseInt(offset, 10) * 30 : 0;
+
+    const query = {
+      where: { [type + 'Id']: id },
+      limit: 30,
+      include: [{ model: Comment, attributes: ['id'] }],
+    };
+
+    if (offset) {
+      query.offset = offset;
+    }
+
+    const comments = await Comment.findAll(query);
+
+    res.json(comments);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.get('/thread/:id', async (req, res, next) => {
+  try {
+    if (!req.params.id) {
+      res.sendStatus(400);
+    }
+
+    const comments = await Comment.findAll({
+      where: { threadId: req.params.id },
+    });
+
+    res.json(comments);
+  } catch (error) {
+    next(error);
+  }
+});
+
+// type: post / event
+router.post('/:type/:typeId/comment/:commentId', isLoggedIn, (req, res) => {
+  const { type, typeId, commentId } = req.params;
+  const { content } = req.body;
+
+  if (!type || !typeId || !commentId || !content) {
+    res.sendStatus(400);
+  }
+
+  const io = req.app.get('io');
+  const room = `${type}-${typeId}`;
+
```

Comment:
I'm uncertain. is req.app.get('io') something that needs to be awaited?

Reply:
I read further down in the code, and I think the answer is no to this question based on how our server is set up now.